Fix and improve image build procedure

ilveroluca opened the pull request:
This  PR changes two things.  The old procedure tried to check out an inexistent tag in the multivariate tool -- it now checks out the `master`.  The second change makes the image build procedure more efficient by doing almost all software installation in a single `RUN` statement, reducing the number of layers in the resulting image.

kikkomep committed:
Correct checkout of inexisting branch

kikkomep committed:
Improve Docker image build

pkrog commented:
The tags have been changed by the tool's author. It is now v2.3.6 instead of 2.3.6. Please put back the right tag instead of master branch, because the author uses this branch for developing, not for actual release (this is something that will hopefully change in the future).

ilveroluca committed:
Correct tool tag to match required version, v2.3.6

ilveroluca committed:
Don't upgrade base image and install everything in a single RUN

Upgrading goes against the best practices.  It results in build that may
not be repeatable and greatly increases the size of the resulting docker
image.

ilveroluca commented:
These latest commits restore the tool version at 2.3.6.  I also got rid of the `upgrade` since upgrading goes against the best practices.  It results in build that may not be repeatable and greatly increases the size of the resulting docker image.  The image has gone from a bit over 600 MB to 329 MB.

ilveroluca commented:
Oh.  I get this warning from R when building the image:

```
package 'ropls' is not available (for R version 3.0.2) 
```
